### PR TITLE
Feature/dimensionality

### PR DESCRIPTION
**Pull request by Ocnarf2070**
Creation of the feature of Dimensionality which implements most of the dimensionality functions of khiva.

### CODE REVIEW

File: aljuarismi/dimensionality.py [hunk 1]
```diff
@@ -7,29 +7,118 @@
...
+        query = al.query_input()
+        while not query.isnumeric():
+            print('Incorrect input.\nIt is not a number.\nPlease introduce one.')
+            al.voice('Incorrect input.\nIt is not a number.\nPlease introduce one.')
+            query = al.query_input()
+        num_points = int(query)
+    return num_points
+
+
+def get_number(parameters):
```
Comment (avilchess) [inline aljuarismi/dimensionality.py]: Why two functions to get a number? one for integers and one for floats?, if that is the case, please put a proper name like get_float_number

File: aljuarismi/dimensionality.py [hunk 2]
```diff
@@ -7,29 +7,118 @@
...
+    k_array = kv.Array(dataset)
+    result = kv.paa(k_array, num_points)
+    return result.to_pandas()
+
+
+def pip(dataset, parameters):
+    """
+    Executes the function pip of khiva.
+    :param dataset: The dataset which will be evaluated.
+    :param parameters: The parameters of the function (number of pip)
```
Comment (avilchess) [inline aljuarismi/dimensionality.py]: lack of point at the end

File: aljuarismi/dimensionality.py [hunk 3]
```diff
@@ -7,29 +7,118 @@
...
+    result.set_index(0, inplace=True)
+    result.set_index(result.index.astype('int32'), inplace=True)
+    return result
+
+
+def ramer_douglas_peucker(dataset, parameters):
+    """
+    Executes the function Ramer-Douglas-Peucker of khiva.
+    :param dataset: The dataset which will be evaluated.
+    :param parameters: The parameters of the function (epsilon)
```
Comment (avilchess) [inline aljuarismi/dimensionality.py]: missing point at the end

File: aljuarismi/dimensionality.py [hunk 4]
```diff
@@ -7,29 +7,118 @@
...
+    result.set_index(0, inplace=True)
+    result.set_index(result.index.astype('int32'), inplace=True)
+    return result
+
+
+def visvalingam(dataset, parameters):
+    """
+    Executes the function Visvalingam of khiva.
+    :param dataset: The dataset which will be evaluated.
+    :param parameters: The parameters of the function (number of points)
```
Comment (avilchess) [inline aljuarismi/dimensionality.py]: missing point at the end

File: aljuarismi/matrix.py [hunk 5]
```diff
@@ -19,6 +19,8 @@ def find_best_n_discords(mt, m, parameters, col, dataset):
     :param mt: The matrix profile and the index calculated previously.
     :param m: Subsequence length value used to calculate the matrix profile.
     :param parameters: The parameters of the function (number of clusters, ...).
+    :param col: The column which has been used in stomp
```
Comment (avilchess) [inline aljuarismi/matrix.py]: missing point

File: aljuarismi/operations.py [hunk 6]
```diff
@@ -10,25 +10,42 @@
 import aljuarismi as al
 
 
-def do_op(parameters):
+def do_dimensionality(parameters):
     """
     Do a operation of dimensionality.
     :param parameters: The parameters of the function (name of the operation, ...).
     :return:
```
Comment (avilchess) [inline aljuarismi/operations.py]: if the function does not return anything delete this line

File: tests/unit_tests/dimensionality_test.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
+import json
```
Comment (avilchess) [inline tests/unit_tests/dimensionality_test.py:10]: missing copyright header

**Review by avilchess — CHANGES_REQUESTED** [review]

File: aljuarismi/dimensionality.py [hunk 8]
```diff
@@ -7,29 +7,118 @@
...
+    :param parameters: the parameters of the function(number,...).
+    :return: An integer number.
     """
-    if timeseries.columns.size > 1:
-        timeseries = timeseries[al.obtain_column(timeseries)]
-    if parameters.get("number") == '':
+    if parameters["number"]:
+        num_points = int(parameters["number"])
+    else:
         print('How many point do you want to reduce it?')
```
Comment (avilchess) [inline aljuarismi/dimensionality.py:24]: What should be the value of epsilon?

File: aljuarismi/dimensionality.py [hunk 9]
```diff
@@ -7,29 +7,118 @@
...
+            query = al.query_input()
+        num_points = int(query)
+    return num_points
+
+
+def get_float_number(parameters):
+    """
+    Obtains or asks for a number.
+    :param parameters: the parameters of the function(number,...).
+    :return: An number.
```
Comment (avilchess) [inline aljuarismi/dimensionality.py]: A float number

File: aljuarismi/dimensionality.py [hunk 10]
```diff
@@ -7,29 +7,118 @@
...
+    """
+    if parameters["number"]:
+        num_points = parameters["number"]
     else:
-        num_points = int(parameters.get("number"))
+        print('How many point do you want to reduce it?')
```
Comment (avilchess) [inline aljuarismi/dimensionality.py]: What should be the number of resulting points?

File: aljuarismi/matrix.py [hunk 11]
```diff
@@ -52,6 +54,8 @@ def find_best_n_motifs(mt, m, parameters, col, dataset):
     :param mt: The matrix profile and the index calculated previously, and the m used.
     :param m: Subsequence length value used to calculate the matrix profile.
     :param parameters: The parameters of the function (number of clusters, ...).
+    :param col: The column which has been used in stomp
```
Comment (avilchess) [inline aljuarismi/matrix.py]: mssing point at the end

File: aljuarismi/utilities.py [hunk 12]
```diff
@@ -45,6 +45,9 @@ def obtain_column(dataset):
 
 
 def query_input():
+    """
+    Execute a query input
```
Comment (avilchess) [inline aljuarismi/utilities.py]: missing point at the end

**Review by avilchess — CHANGES_REQUESTED** [review]

File: aljuarismi/dimensionality.py (hunk 8 above)
Comment (Ocnarf2070) [reply aljuarismi/dimensionality.py:24]: For obtaining the epsilon value, it has his own function which is called get_epsilon, before called get_number or get_float_number

**Review by avilchess — APPROVED** [review]

**Pull request merged**